src/SDL12_compat.c

```diff
@@ -5231,7 +5231,7 @@ SDL_SetVideoMode(int width, int height, int bpp, Uint32 flags12)
     switch (bpp) {
         case  8: appfmt = SDL_PIXELFORMAT_INDEX8; break;
         case 16: appfmt = SDL_PIXELFORMAT_RGB565; FIXME("bgr instead of rgb?"); break;
-        case 24: appfmt = SDL_PIXELFORMAT_RGB24; FIXME("bgr instead of rgb?"); break;
+        case 24: appfmt = SDL_PIXELFORMAT_RGB888; FIXME("bgr instead of rgb?"); break;
```

Comment:
I don't think this is right. The app is asking for a 24-bit surface, and you're returning a 32-bit surface. If it ends up touching the pixels, then it will be very wrong if it's blindly assuming that SDL hasn't changed the format.

@icculus, am I missing something here?

Reply:
I'd have to check, but maybe if asked for 24 bit color you got XRGB back from the X server in SDL 1.2...? It's certainly possible it worked out like that.

Reply:
Yep, that's likely.